File: api/Domain/WorkItem.cs

namespace Domain
{
    public class WorkItem
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string? Description { get; set; }
        public int Order { get; set; }
        public int ProjectColumnId { get; set; }
        public string? AssignedToUserId { get; set; }
        public string AuthorUserId { get; set; }
        public Priority Priority { get; set; } = Priority.Medium;
        public DateOnly? DueDate { get; set; }
        public ItemType Type { get; set; } = ItemType.Task;
    }

    public enum ItemType
    {
        Task = 1,
        Bug = 2,
        Feature = 3
    }
    public enum Priority
    {
        Lowest = 1,
        Low = 2,
        Medium = 3,
        High = 4,
        Highest = 5
    }
}

File: frontend/Client/src/app/core/models/WorkItem.ts
import { ItemType, Priority } from "./Project";

export interface CreateWorkItemDto{
    title:string,
    description:string|null,
    projectcolumnid:number,
    assignedtouserid:string|null,
    assignedtousername:string|null,
    type: number;
    dueDate?: string | null;  
}

export interface ReorderWorkItemDto{
    workItemId: number;
    columnid: number;
    PrevOrder: number|null;
    NextOrder: number|null;
}

export interface ReorderResultDto{
    Order:number;
}

export interface WorkItemDto{
    id:number;
    title:string;
    description:string|null;
    assignedToUserId:string|null;
    assignedToUserName:string|null;
    authorUserId:string;
    authorUserName:string;
    type: ItemType;
    priority: Priority;
    dueDate?: string | null;
}

export interface UpdateWorkItemDto{
    title?:string|null;
    description?:string|null;
    assignedToUserId?:string|null;
    type?: ItemType|null;
    priority?: Priority|null;
    dueDate?: string|null;
}